ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthEstimateGas.java

```diff
@@ -120,18 +137,15 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     return new JsonRpcSuccessResponse(requestContext.getRequest().getId(), Quantity.create(high));
   }
 
-  private Optional<TransactionSimulatorResult> executeSimulation(
-      final BlockHeader blockHeader,
-      final CallParameter modifiedCallParams,
-      final EstimateGasOperationTracer operationTracer,
-      final boolean allowExceedingBalance) {
-    return transactionSimulator.process(
-        modifiedCallParams,
-        ImmutableTransactionValidationParams.builder()
-            .from(TransactionValidationParams.transactionSimulator())
-            .isAllowExceedingBalance(allowExceedingBalance)
-            .build(),
-        operationTracer,
-        blockHeader.getNumber());
+  private long getBlockNumberFromParameter(final BlockParameter blockParameter) {
+    if (blockParameter.isLatest()) {
+      return blockchainQueries.headBlockNumber();
+    } else if (blockParameter.isEarliest()) {
+      return BlockHeader.GENESIS_BLOCK_NUMBER;
+    } else if (blockParameter.isPending()) {
+      return blockchainQueries.headBlockNumber() + 1;
```

Comment:
is this what other methods do for pending? I thought pending had the same behavior as latest elsewhere

Reply:
Yeah, I just noticed that, will change it.